bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/Auditor.java

```diff
@@ -558,14 +556,26 @@ private void scheduleReplicasCheckTask() {
         executor.scheduleAtFixedRate(auditorReplicasCheckTask, initialDelay, interval, TimeUnit.SECONDS);
     }
 
-    private class UnderReplicatedLedgersChangedCb implements GenericCallback<Void> {
-        @Override
-        public void operationComplete(int rc, Void result) {
+    private void scheduleReplicateStatTask() {
+        long interval = conf.getAuditorUnderReplicasStatInterval();
+        if (interval <= 0) {
+            LOG.info("Under Replicas Stats disabled");
+            return;
+        }
+        auditorStatExecutor = Executors.newSingleThreadScheduledExecutor(
+                new ThreadFactory() {
+                    @Override
+                    public Thread newThread(Runnable r) {
+                        Thread t = new Thread(r, "AuditorStat-" + bookieIdentifier);
+                        t.setDaemon(true);
+                        return t;
+                    }
+                });
+        auditorStatExecutor.scheduleAtFixedRate(() -> {
             Iterator<UnderreplicatedLedger> underreplicatedLedgersInfo = ledgerUnderreplicationManager
                     .listLedgersToRereplicate(null);
             auditorStats.getUnderReplicatedLedgersGuageValue().set(Iterators.size(underreplicatedLedgersInfo));
```

Comment:
We update the Gauge metric with a specific interval, and I'm not sure whether it is meaningful for the metric on the dashboard.

Reply:
I think this metric doesn't need to be highly real-time. We just need to give users an approximate value and indicate whether it has changed. If real-time values are required, we can retrieve the current data of under-replicated ledgers using the BookKeeper shell.

Reply:
If it is better to use `bin/bookkeeper shell` to get the count of under-replicated ledgers directly?

Reply:
Yes, it is. If so, we can remove the auditorStatExecutor. If the user wants to know the under-replication ledger stats, use the bookkeeper shell instead of auditor metrics.